Gems/AtomLyIntegration/EditorModeFeedback/Assets/Shaders/EditorModeCommon.azsli

```diff
@@ -76,4 +82,61 @@ partial ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
     {
         return CalculateOutputColor(inColor, finalEffect, PassSrg::CalculateFinalBlendAmount(t));
     }
-}
\ No newline at end of file
+}
+
+//! Calculates the view position and normal from the fragment position.
+void ComputeViewNormalAndPosition(VSOutput IN, out float3 outPositionWS, out float3 outNormalWS)
```

Comment:
It would be a lot cheaper if the vertex shader used just outputs the normal and the position in the `VSOuput`, rather that recalculate everything in the pixel shader with these helpers. Is that not a possibility?

Reply:
I see. Thanks. I'll let the the guys from sig-graphics to take a look and give the second approval.

Reply:
> It would be a lot cheaper if the vertex shader used just outputs the normal and the position in the VSOuput, rather that recalculate everything in the pixel shader with these helpers. Is that not a possibility

This isn't really true universally. Forwarding entries from VS to PS via ISBE (inter-stage buffer entries) allocates from LDS (local data share) storage which has a direct impact on occupancy (similar effect to allocating registers from the RF). In addition, vertices that are occluded are always fully shaded, as opposed to pixels that can be rejected with early-Z. The choice to compute data at vertex rate and pixel rate is, as a result, a more nuanced decision that depends on the geometry and rest of the shader itself. These routines should be orthogonal to that decision.

BTW, I could have sworn these routines already existed, will do some digging.

Reply:
Oh, is this code just duplicated from the fullscreen shadow code implementation?

Reply:
Yeah, it's mostly a cannibalization of some existing code in O3DE. Both versions have their own coupling to their use cases, a generic version could quite easily be refactored and used in both, though 👍

Reply:
The shadow mapping impl has conditional compilation for both regular and compute shaders so it's not going to be able to be separated out into a generic util. I'm gonna stick my refactored version into  `ScreenSpaceUtil.azsli`, though.